src/core/classes/erc-20.ts

```diff
@@ -396,4 +376,9 @@ export class Erc20<T extends TokenERC20> implements UpdateableNetwork {
       BigNumber.from(value),
     );
   }
+
+  protected async normalizeAmount(amount: Amount): Promise<BigNumber> {
+    const decimals = await this.contractWrapper.readContract.decimals();
```

Comment:
maybe over-optimization but I wonder if we can cache this on the class itself (aka after the first time we look it up we just know what it is. decimal setting is not changeable so once we have it for a given contract instance once we should be able to just return it immediately instead of calling the contract)

Reply:
That's a good point. Doesn't hurt to cache this